SwiftCompilerSources/Sources/Optimizer/FunctionPasses/LifetimeDependenceScopeFixup.swift

```diff
@@ -40,93 +40,110 @@ let lifetimeDependenceScopeFixupPass = FunctionPass(
     guard let lifetimeDep = LifetimeDependence(markDep, context) else {
       continue
     }
-    guard let beginAccess = extendAccessScopes(dependence: lifetimeDep,
-                                               context) else {
-      continue
+    if let arg = extendAccessScopes(dependence: lifetimeDep, context) {
+      markDep.baseOperand.set(to: arg, context)
     }
-    extendDependenceBase(dependenceInstruction: markDep,
```

Comment:
Is this not needed anymore?

Reply:
`extendDependenceBase` was a hack to handle the special case in which the dependent value is returned from a function and the access scope is in the caller. This case is handled directly now so we don't need the hack.